### PR TITLE
Update libjpeg-turbo 2.0.0 => 2.0.4

### PR DESCRIPTION
**Pull request by yongtang**
This PR updates libjpeg-turbo from 2.0.0 (July 2018) to the latest 2.0.4 (Dec 2019),
as there are quite a few CVE related vulnerabilities
(https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=libjpeg) since then.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: third_party/jpeg/BUILD.bazel [hunk 1]
```diff
@@ -524,12 +524,12 @@ JCONFIG_NOWIN_COMMON_SUBSTITUTIONS = {
     "#cmakedefine HAVE_STDDEF_H": "#define HAVE_STDDEF_H 1",
     "#cmakedefine HAVE_STDLIB_H": "#define HAVE_STDLIB_H 1",
     "#cmakedefine NEED_SYS_TYPES_H": "#define NEED_SYS_TYPES_H",
-    "#cmakedefine NEED_BSD_STRINGS": "",
+    "#cmakedefine NEED_BSD_STRINGS 1": "",
```
Comment (mihaimaruseac) [inline third_party/jpeg/BUILD.bazel:527]: Should the 1 be moved to the right of `:`, inside the other pair of quotes? Both here and below

Comment (yongtang) [reply third_party/jpeg/BUILD.bazel:527]: @mihaimaruseac In 2.0.0 version of libjpeg-turbo the cmakedefine is:
```
#cmakedefine NEED_BSD_STRINGS
```

So in the past the replacement bazel rule is would be the following to undef:
```
"#cmakedefine NEED_BSD_STRINGS": "",
```

And the following for define:
```
"#cmakedefine NEED_BSD_STRINGS": "#define NEED_BSD_STRINGS 1",
```

However, in 2.0.4 libjpeg-turbo changed the way the template was setup. It is now:
```
#cmakedefine NEED_BSD_STRINGS 1
```

So in order to undef I use the following:
```
"#cmakedefine NEED_BSD_STRINGS 1": "",
```

And I use the following for define (note the extra `1` is actually not touched:
```
"#cmakedefine NEED_BSD_STRINGS": "#define NEED_BSD_STRINGS",
```

**Comment by yongtang** [comment]
@mihaimaruseac I updated the PR to always use the full line (with `1` at the end) for replacement rule in bazel:
```
"#cmakedefine HAVE_STDLIB_H 1": "#define HAVE_STDLIB_H 1",
```

I think this will make the rule clear (and helps in future maintenance). Please take a look.

File: third_party/jpeg/BUILD.bazel (hunk 1 above)
Comment (mihaimaruseac) [reply third_party/jpeg/BUILD.bazel:527]: Thank you for detailing

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**